cmd/prchecker/main.go

```diff
@@ -41,79 +71,163 @@ func main() {
 
 	client := github.NewClient(nil)
 
-	files, err := ListAllPullRequestFiles(client, owner, pullrequest, repo)
+	githubRepo := &GitHubRepository{
+		owner: owner,
+		repo:  repo,
+	}
+
+	// Check if module span is allowed before scanning on commits
+	isSpanAllowed, err := IsModuleSpanAllowed(client, githubRepo, *pullrequest)
 
 	if err != nil {
 		fmt.Println("Unable to retrieve pull request details:", err.Error())
 		os.Exit(2)
 	}
 
-	contributedRestrictedPaths := CountRestrictedPathUses(files, restrictedPaths)
-	modifiedRestrictedDirectories := CountModifiedRestrictedDirectories(contributedRestrictedPaths)
+	if isSpanAllowed {
+		fmt.Println("Check not run. Module spanning was allowed in this Pull Request.")
+		os.Exit(0)
+	}
+
+	isSpanningPull, _, err := IsPullRequestSpanningPathList(client, githubRepo, *pullrequest, restrictedPaths)
+
+	if err != nil {
+		fmt.Println("Unable to retrieve pull request details:", err.Error())
```

Comment:
A common way to print errors is `fmt.Printf("unable to retrieve pull request details: %v", err)`